src/fstests/fstests.jinja2

```diff
@@ -63,7 +73,7 @@ class Job(BaseJob):
         if self._check_kvm_xfstest_env():
             if self._kernel_config(src_path) == 'pass':
                 if self._kernel_build(src_path) == 'pass':
-                    return self._run_smoke_tests()
-        else:
-            return 'fail'
+                    if self._run_smoke_tests() == 'pass':
+                        return self._get_xml_from_vm(src_path)
+        return 'fail'
```

Comment:
We'll have to stop nesting these `if` statements at some point or it'll become very hard to read and maintain.

Reply:
Updated, now we are returning False/True as we discussed.